secure_login_gateway/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use thiserror::Error;
+
+/// Limits for password length
+const MIN_PASS_LENGTH: usize = 8;
+const MAX_PASS_LENGTH: usize = 16;
+
+/// Password possible errors
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum PassError {
+    #[error("Invalid password length, should be between {MIN_PASS_LENGTH} to {MAX_PASS_LENGTH}")]
+    InvalidPassLength,
+
+    #[error("A password must contain at least one small letter")]
+    NoSmallLetter,
+
+    #[error("A password must contain at least one capital letter")]
+    NoCapitalLetter,
+
+    #[error("A password must contain at least one digit")]
+    NoDigit,
+}
+
+pub type PassResult = Result<(), PassError>;
```

Comment:
```suggestion
pub type Result = Result<(), PassError>;
```
And what i gonna do if i want to return something in the ok value? (Make it template)